xarray/core/common.py

```diff
@@ -161,13 +161,15 @@ class AttrAccessMixin:
     _initialized = False
 
     @property
-    def _attr_sources(self):
-        """List of places to look-up items for attribute-style access"""
+    def _attr_sources(self) -> List[Mapping[Hashable, Any]]:
```

Comment:
Is `Mapping[Hashable, Any]` identical to `Mapping`?

Reply:
There is no requirement for the keys of a Mapping to be hashable. This is relevant for the various setter methods Dataset.attrs which cast any Mapping to an OrderedDict.